public/components/metrics/sidebar/data_source_picker.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiComboBox, EuiTitle } from '@elastic/eui';
+import React from 'react';
+import { DATASOURCE_OPTIONS } from '../../../../common/constants/metrics';
+import { OptionType } from '../../../../common/types/metrics';
+
+interface DataSourcePickerMenuProps {
+  selectedDataSource: OptionType[];
+  setSelectedDataSource: (sources: OptionType[]) => void;
+}
+
+export const DataSourcePicker = ({
+  selectedDataSource,
```

Comment:
Since datasource is in metric_slice, this component should be directly using `dispatch` and `useSelector`